src/include/kompute/Tensor.hpp

```diff
@@ -247,7 +241,7 @@ class Tensor
   protected:
     // -------------- ALWAYS OWNED RESOURCES
     TensorTypes mTensorType;
-    TensorDataTypes mDataType;
+    std::shared_ptr<ABCTypeContainer> mDataType;
```

Comment:
Why does this need to be a sharedptr ? Would be keen to avoid needing this complexity for the datatype

Reply:
I've changed it to a regular pointer now and freed the memory when tensor is free.